packages/edit-site/src/components/page-patterns/use-patterns.js

```diff
@@ -155,9 +155,13 @@ const selectPatterns = createSelector(
 		];
 
 		if ( syncStatus ) {
-			patterns = patterns.filter(
-				( pattern ) => pattern.syncStatus === syncStatus
-			);
+			// User patterns can have their sync statuses checked directly
+			// Non-user patterns are all unsynced for the time being.
+			patterns = patterns.filter( ( pattern ) => {
+				return pattern.id
+					? pattern.syncStatus === syncStatus
+					: syncStatus === PATTERN_SYNC_TYPES.unsynced;
```

Comment:
Would it be less confusing if we directly filter the `themePatterns` before concatenating it with `userPatterns` several lines above? `userPatterns` should also be filtered in `selectUserPatterns` too so we probably don't need to do that again.